File: credit-service/internal/model/credit.go
package model

type CreditAccount struct {
	ID        int64   `json:"id"`
	UserID    int64   `json:"user_id"`
	Balance   float64 `json:"balance"`
	Status    string  `json:"status"`
	CreatedAt string  `json:"created_at"`
	UpdatedAt string  `json:"updated_at"`
}

type CreditTransaction struct {
	ID              int64   `json:"id"`
	CreditAccountID int64   `json:"credit_account_id"`
	Type            string  `json:"type"`
	Amount          float64 `json:"amount"`
	ReferenceID     string  `json:"reference_id,omitempty"`
	Details         string  `json:"details,omitempty"`
	SM3Hash         string  `json:"sm3_hash"`
	Status          string  `json:"status"`
	CreatedAt       string  `json:"created_at"`
}

type EarnRequest struct {
	UserID      string  `json:"user_id" binding:"required"`
	Amount      float64 `json:"amount" binding:"required,gt=0"`
	Type        string  `json:"type" binding:"required"`
	ReferenceID string  `json:"reference_id"`
	Details     string  `json:"details"`
}

type ConsumeRequest struct {
	UserID      string  `json:"user_id" binding:"required"`
	Amount      float64 `json:"amount" binding:"required,gt=0"`
	ReferenceID string  `json:"reference_id" binding:"required"`
	Details     string  `json:"details"`
}

type RefundRequest struct {
	UserID      string  `json:"user_id" binding:"required"`
	Amount      float64 `json:"amount" binding:"required,gt=0"`
	ReferenceID string  `json:"reference_id" binding:"required"`
	Details     string  `json:"details"`
}

type CalculateDiscountRequest struct {
	UserID            string  `json:"user_id" binding:"required"`
	SubscriptionPrice float64 `json:"subscription_price" binding:"required,gt=0"`
}

type CalculateDiscountResponse struct {
	AvailableBalance float64 `json:"available_balance"`
	MaxDiscount      float64 `json:"max_discount"`
	RemainingToPay   float64 `json:"remaining_to_pay"`
}

type AccountResponse struct {
	UserID  int64   `json:"user_id"`
	Balance float64 `json:"balance"`
	Status  string  `json:"status"`
}

type TransactionListResponse struct {
	Transactions []CreditTransaction `json:"transactions"`
	Total        int                 `json:"total"`
	Page         int                 `json:"page"`
	PageSize     int                 `json:"page_size"`
}

File: ios/AccountCenter/Models/Credit.swift
import Foundation

struct CreditAccount: Codable {
    let userId: Int64
    let balance: Double
    let status: String

    enum CodingKeys: String, CodingKey {
        case userId = "user_id"
        case balance
        case status
    }
}

struct Transaction: Codable, Identifiable {
    let id: Int64
    let creditAccountId: Int64
    let type: String
    let amount: Double
    let referenceId: String?
    let details: String?
    let status: String
    let createdAt: String

    enum CodingKeys: String, CodingKey {
        case id
        case creditAccountId = "credit_account_id"
        case type
        case amount
        case referenceId = "reference_id"
        case details
        case status
        case createdAt = "created_at"
    }
}

struct TransactionList: Codable {
    let transactions: [Transaction]
    let total: Int
    let page: Int
    let pageSize: Int

    enum CodingKeys: String, CodingKey {
        case transactions
        case total
        case page
        case pageSize = "page_size"
    }
}

struct ReferralSummary: Codable {
    let totalReferees: Int
    let totalEarned: Double
    let activeReferees: Int

    enum CodingKeys: String, CodingKey {
        case totalReferees = "total_referees"
        case totalEarned = "total_earned"
        case activeReferees = "active_referees"
    }
}

struct CalculateDiscountRequest: Codable {
    let userId: String
    let subscriptionPrice: Double

    enum CodingKeys: String, CodingKey {
        case userId = "user_id"
        case subscriptionPrice = "subscription_price"
    }
}

struct DiscountInfo: Codable {
    let availableBalance: Double
    let maxDiscount: Double
    let remainingToPay: Double

    enum CodingKeys: String, CodingKey {
        case availableBalance = "available_balance"
        case maxDiscount = "max_discount"
        case remainingToPay = "remaining_to_pay"
    }
}

struct GenerateLinkRequest: Codable {
    let userId: String

    enum CodingKeys: String, CodingKey {
        case userId = "user_id"
    }
}

struct ReferralLinkData: Codable {
    let referralCode: String
    let referralLink: String

    enum CodingKeys: String, CodingKey {
        case referralCode = "referral_code"
        case referralLink = "referral_link"
    }
}
